main.go

```diff
@@ -667,6 +831,8 @@ func parseFlags() (config, error) {
 		"Path to the RBAC configuration file.")
 	flag.StringVar(&cfg.tenantsConfigPath, "tenants.config", "tenants.yaml",
 		"Path to the tenants file.")
+	flag.IntVar(&cfg.tenantRegistrationRetryCount, "tenants.registration-retry-count", 3,
```

Comment:
I don't think we should have a limit here. Instead we should continuously retry registering tenants and use exponential backoff.

Reply:
👍